### PR TITLE
Adding setText and getText function to Brls::Button

### PR DESCRIPTION
**Pull request by EmreTech**
This Pull Request is for the yoga branch, not the master branch.

In short, make a function in the Brls::button class to change the text of the button.

### CODE REVIEW

File: library/include/borealis/views/button.hpp [hunk 1]
```diff
@@ -163,6 +163,8 @@ class Button : public Box
      */
     void setState(ButtonState state);
 
+    void setText(std::string text);
```
Comment (natinusala) [inline library/include/borealis/views/button.hpp:166]: Can you add a documentation comment please ?

**Review by natinusala — CHANGES_REQUESTED** [review]

File: library/include/borealis/views/button.hpp (hunk 1 above)
Comment (EmreTech) [reply library/include/borealis/views/button.hpp:170]: Alright, done

**Review by EmreTech — COMMENTED** [review]
Don't know if you got my comment yet, but I added a documentation comment

File: demo/components_tab.cpp [hunk 2]
```diff
@@ -33,6 +33,12 @@ ComponentsTab::ComponentsTab()
 bool ComponentsTab::onPrimaryButtonClicked(brls::View* view)
 {
     // TODO: actually do something here
+    primary_button->setText("Some basic text");
```
Comment (natinusala) [inline demo/components_tab.cpp:36]: What is the purpose of that? To test that setText and getText work?

While I like the intent, the demo is not the place for such a test. Actually... there is no place for such a test, as we currently don't have a testing suite with borealis.

I want one but I didn't find something that suited all my needs (aka hook into the tested app, run it for multiple frames and test what's going on _inside_). I think we would need a custom made one, which takes time and effort.

**Review by natinusala — CHANGES_REQUESTED** [review]

File: demo/components_tab.cpp (hunk 2 above)
Comment (EmreTech) [reply demo/components_tab.cpp]: Done

File: library/include/borealis/views/button.hpp [hunk 3]
```diff
@@ -163,6 +163,17 @@ class Button : public Box
...
     void setState(ButtonState state);
 
+
+    /**
+     * Sets the text of the button. 
+     */
+    void setText(std::string text);
+
+    /**
+     * Gets the text of the button and returns it
```
Comment (natinusala) [inline library/include/borealis/views/button.hpp:173]: "gets" is a synonym of "return", there is no need to say "get and return" 😛 

"Returns the text of the button" is fine

Comment (EmreTech) [reply library/include/borealis/views/button.hpp]: Fixed that. I probably should pay more attention to my grammar.

**Pull request merged**